test/test_utils.py

```diff
@@ -258,10 +258,13 @@ def test(self):
         self.assertEqual("0:00", util.format_time(0, display_zero=True))
         self.assertEqual("3:00", util.format_time(179750))
         self.assertEqual("3:00", util.format_time(179500))
+        self.assertEqual("3:00", util.format_time(179500.5))
         self.assertEqual("2:59", util.format_time(179499))
         self.assertEqual("59:59", util.format_time(3599499))
         self.assertEqual("1:00:00", util.format_time(3599500))
         self.assertEqual("1:02:59", util.format_time(3779499))
+        with self.assertRaises(ValueError):
+            util.format_time(-40000)
```

Comment:
Please also test it raises `TypeError` if called with a value than can't be converted to `float`.
Else the PR looks good, maybe squash your commits ;)

Reply:
Settled on float being a valid input (and added a test for that above), no reason not to handle fractional milliseconds any differently to a small number of milliseconds, they all get rounded below 1 second anyway.

Will do on the squash.

Reply:
The test for `TypeError` is still missing:

 ```python
with self.assertRaises(TypeError):
    util.format_time(list())
```

Reply:
Fixed!